tests/MongoDB.Driver.Tests/Specifications/unified-test-format/UnifiedTestFormatTestRunner.cs

```diff
@@ -0,0 +1,214 @@
+                entities: testCase.Shared["createEntities"].AsBsonArray, // TODO: can be absent
+                initialData: testCase.Shared.GetValue("initialData", null)?.AsBsonArray,
+                test: testCase.Test);
+        }
+
+        public void Run(
+            string schemaVersion,
+            BsonArray runOnRequirements,
+            BsonArray entities,
+            BsonArray initialData,
+            BsonDocument test)
+        {
+            schemaVersion.Should().StartWith("1.0");
+            if (runOnRequirements != null)
+            {
+                RequireServer.Check().RunOn(runOnRequirements);
+            }
+
+            _entityMap = new EntityMap(entities);
+
+            if (initialData != null)
+            {
+                AddInitialData(DriverTestConfiguration.Client, initialData);
+            }
+
+            foreach (var operationItem in test["operations"].AsBsonArray)
+            {
+                var operation = CreateOperation(operationItem.AsBsonDocument, _entityMap);
+                var result = ExecuteOperation(operation, test["async"].AsBoolean);
+                if (result != null)
+                {
+                    AssertOperation(operationItem.AsBsonDocument, result, _entityMap);
+                }
+                if (operation is UnifiedFailPointOperation failPointOperation)
+                {
+                    _failPoints.Add(failPointOperation.FailPoint);
+                }
+            }
+
+            if (test.AsBsonDocument.TryGetValue("outcome", out var expectedOutcome))
+            {
+                AssertOutcome(DriverTestConfiguration.Client, expectedOutcome.AsBsonArray);
+            }
+            if (test.AsBsonDocument.TryGetValue("expectEvents", out var expectedEvents))
+            {
+                AssertEvents(expectedEvents.AsBsonArray, _entityMap);
+            }
+        }
+
+        public void Dispose()
+        {
+            foreach (var failPoint in _failPoints)
+            {
+                failPoint?.Dispose();
+            }
+            _entityMap?.Dispose();
+        }
+
+        // private methods
+        private void AddInitialData(IMongoClient client, BsonArray initialData)
+        {
+            foreach (var dataItem in initialData)
+            {
+                var collectionName = dataItem["collectionName"].AsString;
+                var databaseName = dataItem["databaseName"].AsString;
+                var documents = dataItem["documents"].AsBsonArray.Cast<BsonDocument>().ToList();
+
+                if (documents.Count > 0)
+                {
+                    var database = client.GetDatabase(databaseName);
+                    var collection = database
+                        .GetCollection<BsonDocument>(collectionName)
+                        .WithWriteConcern(WriteConcern.WMajority);
+
+                    database.DropCollection(collectionName);
```

Comment:
Question: do we want to drop the collection even if there are no documents to insert? That might be the case if the expected outcome is an empty collection.

If so the call to `DropCollection` needs to move to before the `if` on line 108.